flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/calcite/DefaultPlannerConfig.scala

```diff
@@ -27,22 +27,15 @@ import org.apache.calcite.sql.parser.SqlParser
 import org.apache.calcite.sql.util.ChainedSqlOperatorTable
 import org.apache.calcite.sql2rel.SqlToRelConverter
 import org.apache.calcite.tools.{RuleSet, RuleSets}
+import org.apache.flink.table.api.PlannerConfig
 import org.apache.flink.util.Preconditions
 
 import scala.collection.JavaConverters._
 
 /**
-  * Builder for creating a Calcite configuration.
-  *
-  * @deprecated This class will be removed temporarily while the API is uncoupled
-  *             from Calcite. An alternative might be provided in the future. See FLINK-11728
-  *             for more information.
+  * Builder for creating a Default Planner configuration.
   */
-@Deprecated
-@deprecated(
-  "This method will be removed temporarily while the API is uncoupled from Calcite.",
-  "1.8.0")
-class CalciteConfigBuilder {
```

Comment:
How about we leave the old name? I think the reason for renaming it was that in the future this config could evolve into more complex planner specific config, right? 

I think though we should try to modularize the config a bit, I think we could take the approach as in the comment on `PlannerConfig`